docs/tutorials/kubernetes-hpa.md

```diff
@@ -0,0 +1,313 @@
+
+### Install OpenFaaS
+
+Use the Deployment guide or an existing installation, you should also install the CLI.
+
+### Install the metrics server
+
+HPAv2 relies on the [Kubernetes metrics-server](https://github.com/kubernetes-incubator/metrics-server) which can be installed using a helm chart.
+
+Most cloud providers are compatible with the metrics-server, but some are not, or require an additional "insecure" flag to be configured.
+
+```sh
+helm install --name metrics-server --namespace kube-system \
+  stable/metrics-server
+```
+
+You can see the [Helm chart](https://github.com/helm/charts/tree/master/stable/metrics-server) for more options.
+
+## 2. Check your metrics-server
+
+When you see metrics appear from this command, continue with the tutorial.
+
+```sh
+kubectl get --raw "/apis/metrics.k8s.io/v1beta1/nodes"
+
+kubectl get --raw /metrics
+```
+
+Check the logs of the metrics-server for any errors:
+
+```
+kubectl logs deploy/metrics-server -n kube-system
+```
+
+If you don't see any metrics, then you may be using a cloud which needs the "insecure" work-around.
+
+```sh
+helm del --purge metrics-server
+
+helm install --name metrics-server --namespace kube-system \
+  stable/metrics-server \
+  --set args="{--kubelet-insecure-tls,--kubelet-preferred-address-types=InternalIP,ExternalIP,Hostname}"
+```
+
+You will now get Pod metrics along with Node metrics, these take a while to propagate.
+
+Find out the CPU and memory usage across nodes:
+
+```sh
+kubectl top node
+NAME   CPU(cores)   CPU%   MEMORY(bytes)   MEMORY%   
+nuc7   377m         4%     7955Mi          24%
+```
+
+Find out the usage across Pods:
+
+```sh
+# View the functions
+kubectl top pod -n openfaas-fn
+NAME                        CPU(cores)   MEMORY(bytes)   
+nodeinfo-6f48f9b548-gbtr4   2m           3Mi
+
+# View the core services
+kubectl top pod -n openfaas
+NAME                                 CPU(cores)   MEMORY(bytes)   
+alertmanager-666c65c694-k8q6h        2m           10Mi            
+basic-auth-plugin-6d97c6dc5b-rbw29   1m           4Mi             
+faas-idler-67f9dcd4fc-85rbt          1m           4Mi             
+gateway-7c687d498f-nvjh8             4m           21Mi            
+nats-d4c9d8d95-fxrc2                 1m           6Mi             
+prometheus-549c7d687d-zznmq          9m           41Mi            
+queue-worker-544bcb7c67-72l72        1m           3Mi
+```
+
+> Note: you can view function usage using `-n openfaas-fn`, or core-service usage with `-n openfaas`
+
+### Disable auto-scaling with OpenFaaS
+
+Disable auto-scaling by scaling alertmanager down to zero replicas, this will stop it from firing alerts.
+
+```sh
+kubectl scale -n openfaas deploy/alertmanager --replicas=0
+```
+
+### Configure Prometheus to scrape the Kubernetes API server
```

Comment:
This not needed, metrics server collects it's own metrics.

Reply:
Will we need it for the custom metrics + adapter later?

Reply:
No, the prom adapter should target the OF Prometheus, no need to install a 2nd one.